MOB-38392: Получение данных с GitHub

VDA22 opened the pull request.

VDA22 committed:
Добавлена логика для запроса данных с GitHub

timbaev commented on Sources/FigmaGen/Commands/TokensCommand.swift:
```diff
@@ -25,10 +25,31 @@ final class TokensCommand: AsyncExecutableCommand {
     )
 
+    let remoteFileOwnerKey = Key<String>(
+        "--remoteFileOwnerKey",
```
Наверное, просто `--remoteFileOwner`, это же не ключ? Или ключ?

timbaev replied:
В других параметрах аналогично

VDA22 replied:
поправила

timbaev commented on Sources/FigmaGen/Generators/Tokens/GenerationParametersResolver/DefaultTokensGenerationParametersResolver.swift:
```diff
@@ -21,19 +21,37 @@ final class DefaultTokensGenerationParametersResolver: TokensGenerationParameter
+            )
+        } else {
+            throw GenerationParametersError.invalidFileConfiguration
+        }
```
Есть вот такой варик, чтобы назначить значения напрямую без `if-else`:
```swift
let file = try configuration.file.map { fileConfiguration in
    guard let accessToken = accessTokenResolver.resolveAccessToken(from: configuration.accessToken) else {
        throw GenerationParametersError.invalidAccessToken
    }

    return FileParameters(
        key: fileConfiguration.key,
        version: fileConfiguration.version,
        accessToken: accessToken
    )
}

let remoteFile = try configuration.remoteRepoConfig.map { remoteFileConfiguration in
    guard let accessToken = remoteFileConfiguration.accessToken else {
        throw GenerationParametersError.invalidAccessToken
    }

    return RemoteFileParameters(
        owner: remoteFileConfiguration.owner,
        repo: remoteFileConfiguration.repo,
        branch: remoteFileConfiguration.branch,
        filePath: remoteFileConfiguration.filePath,
        accessToken: accessToken
    )
}

if file.isNil && remoteFile.isNil {
    throw GenerationParametersError.invalidFileConfiguration
}
```

VDA22 replied:
обновила

timbaev commented on Sources/FigmaGen/Providers/FigmaAPI/DTOs/File/Nodes/GitHubFile.swift:
Это точно не должно быть в папочке FigmaAPI, потому что к нему не относится
Скорее всего лучше рядом создать GitHubAPI и положить туда

VDA22 replied:
перенесла в `GitHubAPI`

timbaev commented on Sources/FigmaGen/Providers/FigmaAPI/GitHubAPIProviderImp.swift:
```diff
@@ -0,0 +1,142 @@
+import PromiseKit
+import FigmaGenTools
+
+final class GitHubAPIProviderImp: RemoteRepoProvider {
```
Кажется, можно просто `GitHubAPIProvider`

VDA22 replied:
тоже обновила

VDA22 committed:
Изменения по комментариям

- убрала лишние key
- обновила конструктор в resolveGenerationParameters
- перенесла фалы для GitHub

VDA22 committed:
Перенесла GitHubFile в парку GitHubApi
